src/Modal.jsx

```diff
@@ -91,16 +91,27 @@ function Modal({
 
   // Handle changing document.body styles based on isOpen state
   useEffect(() => {
-    if (isOpen && !allowScroll) {
-      prevBodyOverflowStyle.current = document.body.style.overflow;
-      document.body.style.overflow = "hidden";
-    }
-
-    return () => {
-      if (!allowScroll) {
-        document.body.style.overflow = prevBodyOverflowStyle.current || "";
-      }
-    };
+     if (isOpen && !allowScroll) {
+       prevBodyStyle.current = {
+         width: document.body.style.width,
+         position: document.body.style.position,
+         overflowY: document.body.style.overflowY
+       };
+       document.body.style.position = "fixed";
+       document.body.style.overflowY = "scroll";
+       document.body.style.width = "100%";
```

Comment:
Can you explain why we need to adjust the width? I saw in the SO post that it's required, but I don't understand why.

Reply:
yeah, tbh not sure also, checked it out and it does prevent reflow changes. Guessing it has something to do with fixed taking the element out of the flow, so we need to readjust it to take 100% width.